Dockerfile

```diff
@@ -5,6 +5,7 @@ RUN apk add --no-cache ca-certificates tzdata && update-ca-certificates
 
 # Install the required packages
 RUN pip install --no-cache-dir redis flower
+RUN pip install --no-cache-dir "SQLAlchemy>=1.4,<2" "psycopg2-binary>=2.9,<3"
```

Comment:
I don't love that we're choosing a version of these libraries to support when building the docker image; that seems like a task that should be left up to individual services to do. For example, what if one microservice wants to use SQLAlchemy 2.x.x, but another wants to use SQLAlchemy 1.4.x? Because Flower interacts with the same database as gets written by the microservice's own Celery worker, it seems like the service should be managing the SQLAlchemy/Postgres library versions. If we want to adopt this, then we may need to shift our pattern away from building a final production Flower image, and instead having a "base" Flower image that other repos will extend in order to render their own results. Does that sound right? 

In the meantime, I don't think we're deviating too far from existing Flower patterns by putting this here. The base branch of Flower already does `pip install ... redis ...`. This introduces the exact same potential for dependency conflicts as I described above, but for Redis library versions instead of SQLAlchemy library versions.

Reply:
> Does that sound right?

Doesn't sound right to me. From what I have seen, Flower is designed to run as a standalone application. This means that whatever service celery backend backend happens to be writing to a Postgres database and Flower happens to be reading from that same Postgres database. In this scenario it shouldn't even matter whether or not Flower is using SQLALchemy. The thing that matters is that the schema that Flower is reading from are the same _schemas_ that Celery is writing to.

> it seems like the service should be managing the SQLAlchemy/Postgres library versions

It's more like that the service and Flower need to agree on the underlying Postgres schema representation, which likely means they need to coordinate the same version of Celery.

Reply:
ship

I'm going to cut this comment short:

I want to touch on matrix testing, 
certainly i would like to know with confidence that that shipped code is going to work with the known support target versions

this gets into a broader discussion of "how do we organize the manpower required to support a rolling release" though, especially because our customers (other devs) are going to largely be expecting to get the latest versions of libx,liby from pypi, enforced by dependabot even

for now, this is ok as is, maybe a todo comment to make sure we verify it works on versions matching celery latest, stable, oldstable versions?

Reply:
I know its a dead conversation, but the phrase

> happens to be writing to a Postgres database and Flower happens to be reading from that same Postgres database

screams red-flag to me, please remind me, why are we not adding api methods to _celery_ that do the database reads? and reading that new api in flower?

Reply:
> why are we not adding api methods to celery that do the database reads? and reading that new api in flower?

We are using the celery API methods through the Celery result backend.

Reply:
oh? what api methods are we using? i thought the whole reason this was being authored was that celery didn't have any api methods to get the data we wanted without us implementing an in-flight task db. this is unrelated to that?

Reply:
The API methods we are using is the celery functionality of a result backend.

_celery_ has a way to get at the data we want, via its result backend. _flower_ does not have a way to read from the celery result backend and visualize it in a webpage alongside all of the other celery information (like the in-flight tasks). That’s what @MaxwellPayne is adding here - capabilities in Flower to read from the celery result backend, with the end goal of contributing it back to Flower.

Reply:
that means celery doesn't have _an api_ to get what we want, and api would look like `curl $celery_endpoint/tasks`. what is the interface we are connecting to to subscribe to celery events? if its rabbitmq, we could get duplex communication by introducing a command queue to celery but, i really wouldn't promote doing IPC like that, HTTP rest is much nicer

Reply:
Celery does not have _an http API_ that is correct.

celery has a Python API used to query from a result store.

we are not subscribing to celery events, celery itself subscribed to its own events and stored them in its result backend.

this conversation is starting to devolve, I feel like we have been over these points before.

Reply:
these points are exactly why this change is a red-flag

Reply:
FYI @MaxwellPayne , @dylan-shipwell and I talked about this offline - we're going to move forward here but we need to write up the tradeoffs we're making and have a diagram of how the Flower deployable with these customizations interact with the Celery result backend. I was going to take a first stab at that and the 3 of us can review to make sure we have the risks and issues fully documented and identified.

Reply:
Agreed. I think in this situation, there is no "right answer." So we should disagree and commit, and write down the reasons why we're making the choices. Do you have a ticket for this that we can reference for posterity?

Reply:
@MaxwellPayne I created https://shipwell.atlassian.net/browse/SHARE-2822, isn't there an epic you have going somewhere for these celery tasks already?

Reply:
Thanks. 🙏 Just updated the ticket with the relevant epic.